ncclient/transport/ssh.py

```diff
@@ -406,17 +406,16 @@ def _auth(self, username, password, key_filenames, allow_agent,
         saved_exception = None
 
         for key_filename in key_filenames:
-            for cls in (paramiko.RSAKey, paramiko.DSSKey, paramiko.ECDSAKey, paramiko.Ed25519Key):
-                try:
-                    key = cls.from_private_key_file(key_filename, password)
-                    self.logger.debug("Trying key %s from %s",
-                                      hexlify(key.get_fingerprint()),
-                                      key_filename)
-                    self._transport.auth_publickey(username, key)
-                    return
-                except Exception as e:
-                    saved_exception = e
-                    self.logger.debug(e)
+            try:
+                key = paramiko.PKey.from_path(key_filename, password.encode("utf-8"))
```

Comment:
Instead of utf8-encoding the password here (which is what Paramiko does internally), we could also call `paramiko.util.b`:

```python
key = paramiko.PKey.from_path(key_filename, paramiko.util.b(password))
```